src/shared/codelens/typescriptCodeLensProvider.ts

```diff
@@ -5,171 +5,113 @@
-import { defaultMetricDatum, registerCommand } from '../telemetry/telemetryUtils'
-import { TypescriptLambdaHandlerSearch } from '../typescriptLambdaHandlerSearch'
-import { LambdaLocalInvokeArguments, LocalLambdaRunner } from './localLambdaRunner'
-
-export class TypescriptCodeLensProvider implements vscode.CodeLensProvider {
-    public onDidChangeCodeLenses?: vscode.Event<void> | undefined
-
-    public async provideCodeLenses(
-        document: vscode.TextDocument,
-        token: vscode.CancellationToken
-    ): Promise<vscode.CodeLens[]> {
 
-        const logger: Logger = getLogger()
-        const search: TypescriptLambdaHandlerSearch = new TypescriptLambdaHandlerSearch(document.uri)
-        const handlers: LambdaHandlerCandidate[] = await search.findCandidateLambdaHandlers()
-
-        const lenses: vscode.CodeLens[] = []
+import { TypescriptLambdaHandlerSearch } from '../typescriptLambdaHandlerSearch'
 
-        handlers.forEach(handler => {
-            const range: vscode.Range = new vscode.Range(
-                document.positionAt(handler.positionStart),
-                document.positionAt(handler.positionEnd),
+async function getSamProjectDirPathForFile(filepath: string): Promise<string> {
+    const packageJsonPath: string | undefined = await findFileInParentPaths(
+        path.dirname(filepath),
+        'package.json'
+    )
+    if (!packageJsonPath) {
+        throw new Error( // TODO: Do we want to localize errors? This might be confusing if we need to review logs.
+            localize(
+                'AWS.error.sam.local.package_json_not_found',
+                'Unable to find package.json related to {0}',
+                filepath
             )
-            const workspaceFolder:
-                vscode.WorkspaceFolder | undefined = vscode.workspace.getWorkspaceFolder(document.uri)
-
-            if (!workspaceFolder) {
-                throw new Error(`Source file ${document.uri} is external to the current workspace.`)
-            }
-
-            lenses.push(this.generateLocalInvokeCodeLens(document, range, handler.handlerName, false, workspaceFolder))
-            lenses.push(this.generateLocalInvokeCodeLens(document, range, handler.handlerName, true, workspaceFolder))
+        )
+    }
 
-            try {
-                lenses.push(this.generateConfigureCodeLens(document, range, handler.handlerName, workspaceFolder))
-            } catch (err) {
-                const error = err as Error
+    return path.dirname(packageJsonPath)
+}
 
-                logger.error(
-                    `Could not generate 'configure' code lens for handler '${handler.handlerName}': `, error
-                )
+export function initialize({
+    configuration,
+    toolkitOutputChannel,
+    processInvoker = new DefaultSamCliProcessInvoker(),
+    taskInvoker = new DefaultSamCliTaskInvoker()
+}: CodeLensProviderParams): void {
+    vscode.commands.registerCommand(
```

Comment:
Restore the new telemetry-based `registerCommand` call

Reply:
✔️